hal_st/stm32fxxx/I2cStm.hpp

```diff
@@ -37,6 +39,8 @@ namespace hal
         virtual void BusError();
         virtual void ArbitrationLost();
 
+        void ReportError(Result result);
```

Comment:
I would consider adding a comment here, since ReportError is not invoked by I2cStm itself, but is there for the purpose of descending implementations of DeviceNotFound, BusError, ArbitrationLost to clean up and continue. Or, as discussed, implement the more beautiful but more involved error policy :)